flink/src/main/java/org/apache/iceberg/flink/sink/FlinkSink.java

```diff
@@ -153,6 +158,11 @@ public Builder hadoopConf(Configuration newHadoopConf) {
       return this;
     }
 
+    public Builder flinkConf(org.apache.flink.configuration.Configuration config) {
+      this.flinkConf = config != null ? config : new org.apache.flink.configuration.Configuration();
+      return this;
+    }
```

Comment:
By allowing user's to pass in `null` flink configuration objects by silently instantiating a new one, are we making it harder for developers to tell if their configuration they've set up elsewhere is being respected? For example, if at the beginning of my program I used `ParamaterTool` somehow to get my configuration - e.g. reading from a file via `ParameterTool.fromPropertiesFile(file)`, or possibly from CLI args via `ParameterTool.fromArgs(args)`, or one of the several ways to parse configuration properties and parameters view `ParameterToool`, would this instantiation of a new config pick up all of these changes?

If not, I think we should disallow null here - it really doesn't make sense to me to call this method with null. In my opinion, calling a builder with null for the configuration option is typically a bug. And it seems like a potential cause of difficult to track bugs for people who are less aware of the complexities of Flink's configuration object and the various ways it can be derived (via `set`, via `ParameterTool`, via `flink-conf.yaml`, via table catalog properties, etc).

However, it's possible I'm mistaken and that by this point in generation of the job graph that new Configuration will somehow have hooked into everything that's needed to get the data however the user intended. And I will admit it's possible that I am mixing up `Configuration` with several related, but separate, entities. If so, please let me know and I do apologize.

Reply:
cc @JingsongLi @openinx

Reply:
Thanks Kyle, I agree with you, we should not use a null to call a builder, someone who want to use this may always know to  pass a defined configuration.

Reply:
Do we need to pass the whole flink configuration to `IcebergStreamWriter` and `IcebergFilesCommitter` ?  Here we only need the `flushInterval`  to emit the data files to downstream , how about just pass the interval value ?   and we could have a default value inside this builder.

Reply:
I think it's better here to pass the conf If we want to add more parameters to control the behaviors, although currently we only need an interval here.